File: domain/src/main/java/com/ykb/app/cryptotrader/domain/component/Logger.java
package com.ykb.app.cryptotrader.domain.component;

import com.ykb.app.cryptotrader.data.dao.ErrorLogDao;
import com.ykb.app.cryptotrader.data.model.ErrorLog;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Autowired;

public class Logger {

    private final org.slf4j.Logger log;

    @Autowired
    private ErrorLogDao errorLogDao;

    private Logger(Class<?> clazz) {
        log = LoggerFactory.getLogger(clazz);
    }

    public static Logger get(Class<?> clazz) {
        return new Logger(clazz);
    }

    public void info(String msg) {
        log.info(msg);
    }

    public void error(String msg) {
        log.error(msg);
        errorLogDao.save(new ErrorLog(msg));
    }

    public void error(String msg, Throwable t) {
        log.error(msg, t);
        if(t instanceof Exception)
            errorLogDao.save(new ErrorLog(msg, (Exception) t));
    }

    public void error(Throwable t) {
        log.error(t.getMessage(), t);
        if(t instanceof Exception)
            errorLogDao.save(new ErrorLog((Exception) t));
    }

}

File: library/utils/src/main/kotlin/com/ykb/app/cryptotrader/utils/logger/Logger.kt
package com.ykb.app.cryptotrader.utils.logger

import com.ykb.app.cryptotrader.utils.logger.data.ErrorLog
import com.ykb.app.cryptotrader.utils.logger.data.ErrorLogRepo
import org.slf4j.LoggerFactory
import org.springframework.beans.factory.annotation.Autowired
import kotlin.reflect.KClass

class Logger(
    clazz: KClass<*>
) {

    @Autowired
    private lateinit var errorLogRepo: ErrorLogRepo

    companion object {
        fun getLogger(clazz: KClass<*>): Logger {
            return Logger(clazz)
        }
    }

    private val log = LoggerFactory.getLogger(clazz.java)

    fun info(msg: String) {
        log.info(msg)
    }

    fun error(msg: String) {
        log.error(msg)
    }

    fun error(msg: String, t: Throwable) {
        log.error(msg, t)
        errorLogRepo.save(ErrorLog(t, msg))
    }

    fun error(t: Throwable) {
        log.error(t.message, t)
        errorLogRepo.save(ErrorLog(t))
    }

}